src/lib/core.js

```diff
@@ -112,43 +73,127 @@ class Core extends EventEmitter {
   }
 
   /**
-   * Used by a client to populate remote routes. Connectors should use the
-   * RoutingTables API directly.
-   *
-   * @param {IlpAddress} destinationLedger (e.g. "us.fed.wf.")
-   * @param {String} nextHop
-   * @param {IlpAddress} connectorSourceAddress (e.g. "us.fed.wf.mark")
+   * @param {Object} query
+   * @param {String} query.sourceAddress Sender's address
+   * @param {String} query.destinationAddress Recipient's address
+   * @param {String} [query.sourceAmount] Either the sourceAmount or destinationAmount must be specified
+   * @param {String} [query.destinationAmount] Either the sourceAmount or destinationAmount must be specified
+   * @param {String|Number} [query.sourceExpiryDuration] Number of seconds between when the source transfer is proposed and when it expires.
+   * @param {String|Number} [query.destinationExpiryDuration] Number of seconds between when the destination transfer is proposed and when it expires.
+   * @param {Object} [query.destinationPrecisionAndScale]
+   * @returns {Promise<Quote>}
    */
-  addRoute (destinationLedger, nextHop, connectorSourceAddress) {
-    this.tables.addLocalRoutes([{
-      source_ledger: '*',
-      destination_ledger: destinationLedger,
-      connector: nextHop,
-      source_account: connectorSourceAddress,
-      points: [[0, 0], [1, 1]]
-    }])
+  quote (query) {
+    return co(this._quote.bind(this), query)
+  }
+
+  * _quote (query) {
+    const hop = this._findBestHopForAmount(
+      query.sourceAddress, query.destinationAddress,
+      query.sourceAmount, query.destinationAmount)
+    if (!hop) return null
+
+    const sourceLedger = hop.sourceLedger
+    const connectorAccount = this.getPlugin(sourceLedger).getAccount()
+    const sourceExpiryDuration = parseDuration(query.sourceExpiryDuration)
+    const destinationExpiryDuration = (sourceExpiryDuration || query.destinationExpiryDuration)
+      ? parseDuration(query.destinationExpiryDuration) : 5
+    const destinationPrecisionAndScale = query.destinationPrecisionAndScale || {}
+    const quote = {connectorAccount, sourceLedger}
+
+    // If we know a route to the destinationAmount, proceed. Otherwise, ask a
```

Comment:
Typo, you mean a route to the `destinationAddress`.

Also, we should probably say "local route" instead of just "route" to make it clearer.